src/Build.UnitTests/Evaluation/Expander_Tests.cs

```diff
@@ -4385,6 +4385,14 @@ public void PropertyFunctionCheckFeatureAvailability(string featureName, string
         [Theory]
         [InlineData("\u3407\ud840\udc60\ud86a\ude30\ud86e\udc0a\ud86e\udda0\ud879\udeae\u2fd5\u0023", 0, 3, "\u3407\ud840\udc60\ud86a\ude30")]
         [InlineData("\u3407\ud840\udc60\ud86a\ude30\ud86e\udc0a\ud86e\udda0\ud879\udeae\u2fd5\u0023", 2, 5, "\ud86a\ude30\ud86e\udc0a\ud86e\udda0\ud879\udeae\u2fd5")]
+// in net742 parts of one emoji are treated as separate graphemes (text elements) while for newer .net they are one grapheme, see doc with change log:
+// https://learn.microsoft.com/en-us/dotnet/core/compatibility/globalization/5.0/uax29-compliant-grapheme-enumeration?WT.mc_id=DOP-MVP-5002735#change-description
+#if NET5_0_OR_GREATER
```

Comment:
This kind of behavior difference seems bad if we're using it to create a filename and the user is alternating between `dotnet build` and VS. Can we avoid it?

Reply:
There is `System.Text.Runes` but I am not sure yet if it fulfills our expectations and does not have the same problem. I'll check it out.

Reply:
Since Full FW doesn't fully observe unicode standard - we'd need to reimplement parts of it ourselves, or use namers unrelated to filenames.
Neither of those two look good :-/

Reply:
Offline suggestion from @rainersigwald - since there should be 1 marker per obj - we can replace problematic chars